### PR TITLE
Update FHIRRouter Azure Events to Store messageID

### PR DESCRIPTION
**Pull request by JFisk42**
This PR adds bundle.identifier to the Azure Custom Events logging. 

Test Steps:
1. Setting local Azure events logging, see [here](https://github.com/CDCgov/prime-reportstream/blob/platform/josh/14332-FHIRRouter-store-messageID/prime-router/docs/observability/local-dev-setup.md) for steps.
2. Post a report that will make it through the pipeline. Easiest way may be through running the end to end test: `./gradlew testEnd2EndUP`
3. Log into Azure with your SU account, navigate to `pdhtest-appinsights`, and open up the logs section.
4. Run the query `customEvents | where name == "ReportRouteEvent" or name == "ReportAcceptedEvent"`
5. Ensure that logs from your computer are being forwarded by looking at the cloud_RoleInstance or city/state columns.
6. Verify that MessageId is being appended to results in customDimensions. 
  a. The MessageId can be specifically pulled into it's own column for easier viewing: `customEvents | where name == "ReportRouteEvent" or name == "ReportAcceptedEvent" | extend MessageId = parse_json(tostring(customDimensions.messageId))`

## Changes
- Added bundle.identifier as "MessageId" to ReportAcceptedEvent and ReportRouteEvent
- Updated two FHIR data files with invalid FHIR syntax (Ex: `"meta.lastUpdated":"2023-09-29T07:57:58.484-07:00","identifier.system":"https://reportstream.cdc.gov/prime-router","identifier.value":"371784"`)
- Updated docs in a few places

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [x] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/14332

### CODE REVIEW

**Comment by arnejduranovic** [comment]
@victor-chaparro just FYI this change is coming.

File: prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt:32]: Should we mention MSH-10 here at all? Are we going to update this comment when we add support for other formats? My guess is no.

File: prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt:34]: nit: Should logic to perform operations on FHIR bundles be moved outside of AzureEventUtils? We've already established this pattern here so that's why this is a "nit".

**Review by arnejduranovic — COMMENTED** [review]
Looks good to me! Will approve once someone runs the tests locally and approves.

File: prime-router/docs/getting-started/running-tests.md
Comment (thetaurean) [inline prime-router/docs/getting-started/running-tests.md:1]: Nice! Thanks for updating these and adding those helpful usages.

File: prime-router/docs/getting-started/running-tests.md
Comment (thetaurean) [inline prime-router/docs/getting-started/running-tests.md:29]: Could we also add a note on running test groups a la:

```
./prime test --run=server2serverauth
```

(or with its gradlew equivalent primeCLI).

File: prime-router/docs/observability/local-dev-setup.md
Comment (thetaurean) [inline prime-router/docs/observability/local-dev-setup.md:37]: On my mac I ran into an issue with opening the jar, might be nice to add a note:

If you encounter an issue with java unable to open the JAR:
- Double-check your path in `settings.local.json` the path should be in escaped quotes \"/path/to/file\"
- Double-click the jar in finder, then go to System Settings -> Security and click "Open Anyway".
  - Alternative is to use the cli tool to remove the quarantine attr, something like this ```xattr -dr com.apple.quarantine <path to file>/my-binary-amd64```
- Redownload the .jar file (you can attempt extracting it with the `unzip` cli command)

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRDestinationFilter.kt
Comment (thetaurean) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRDestinationFilter.kt:1]: I will update the `FHIRReceiverFilter` following this pattern.

File: prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt
Comment (thetaurean) [inline prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt:347]: nit: collapse to one line

File: prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt
Comment (thetaurean) [inline prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt:450]: nit: Any reason not to set this directly with the preceding when statement (line 441/444)

**Review by thetaurean — APPROVED** [review]
Looks awesome! 👍🥇  I was able to test everything out locally and verify that it showed up in the pdhtest instance.

I left a myriad of nitpicks and one minor coment.

File: prime-router/docs/observability/local-dev-setup.md
Comment (JFisk42) [reply prime-router/docs/observability/local-dev-setup.md:38]: Added

File: prime-router/docs/getting-started/running-tests.md
Comment (JFisk42) [reply prime-router/docs/getting-started/running-tests.md]: This comment inspired me to look over this doc file with fresh eyes and I ended up making a host of changes. If you have a moment to give this doc another once over as a sanity check that would be great: https://github.com/CDCgov/prime-reportstream/blob/platform/josh/14332-FHIRRouter-store-messageID/prime-router/docs/getting-started/running-tests.md

File: prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt
Comment (JFisk42) [reply prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt]: Thanks for pointing this out. Unnecessary line expansion is a pet peeve of mine!

File: prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt
Comment (JFisk42) [reply prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt]: No that I could find, done!

File: prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt
Comment (JFisk42) [reply prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt]: Modified to `Returns the bundle identifier elements that relate to HL7v2 Message Header information for tracking` let me know if this is not general enough. (After pushing it I'm now thinking maybe some HL7 reports get this from something entirely different than message header, no idea).

File: prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt
Comment (JFisk42) [reply prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt:34]: I did go back and forth on this a little, for the moment I do like it here with the other Azure bundle operation. Just to note for the future: I could really only find one other use of the bundle identifier ([here](https://github.com/CDCgov/prime-reportstream/blob/095f3ecd8e8a1265abd63560979e0172474b0ee8/prime-router/src/main/kotlin/fhirengine/engine/ProcessedItem.kt#L31-L33)). I think if bundle.identifier gets used again, moving all of this to FHIRBundleHelpers.kt becomes more necessary.

File: prime-router/docs/getting-started/running-tests.md
Comment (thetaurean) [reply prime-router/docs/getting-started/running-tests.md]: Looks awesome 👍

**Review by arnejduranovic — APPROVED** [review]
great work! Code changes look good to me.

**Pull request merged**
